src/main/java/com/kryptokrauts/aeternity/sdk/service/transaction/impl/TransactionServiceImpl.java

```diff
@@ -173,4 +187,46 @@ private Tx createGeneratedTxObject(String signedTx) {
     tx.setTx(signedTx);
     return tx;
   }
+
+  private PostTransactionResult waitUntilTransactionIsMined(
+      PostTransactionResult postTransactionResult) {
+    if (postTransactionResult != null) {
+      String transactionHash = postTransactionResult.getTxHash();
+      int currentBlockHeight = -1;
+      GenericSignedTx minedTransaction = null;
+      int elapsedTrials = 1;
+      while (currentBlockHeight == -1
+          && elapsedTrials < this.config.getNumTrialsToWaitForTxMined()) {
+        minedTransaction = this.externalApi.rxGetTransactionByHash(transactionHash).blockingGet();
+        if (minedTransaction.getBlockHeight().intValue() > 1) {
+          _logger.debug("Transaction is mined - " + minedTransaction.toString());
+          currentBlockHeight = minedTransaction.getBlockHeight().intValue();
+        } else {
+          _logger.info(
+              String.format(
+                  "Transaction not mined yet, checking again in %s seconds (trial %s of %s)",
```

Comment:
I think we should convert the seconds value to a double here just to make sure we always display the correct value

Reply:
Print is adapted to display exact seconds value